examples/user_guide/04_Machine_Learning.ipynb

```diff
@@ -4,8 +4,485 @@
+   "outputs": [],
+   "source": [
+    "sample_by_feature = flattened_by_band.transpose('z', 'band')\n",
+    "sample_by_feature"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we have the data in the shape that we are looking for: a long array of pixels for each band. As a sanity check we can take a look at the plain `np.array`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X = sample_by_feature.values\n",
+    "X"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Other preprocessing\n",
+    "\n",
+    "Sometimes values are too big or need more axes, or need to have a affine transformation applied. Here we'll demonstrate doing this in `numpy` or `xarray`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Add an axis:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.expand_dims(X, 2).shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_by_feature.expand_dims(dim='e', axis=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Rescale:\n",
+    "\n",
+    "Rescale the data to input to the algorithm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(X - X.mean()) / X.std()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rescaled = (sample_by_feature - sample_by_feature.mean()) / sample_by_feature.std()\n",
+    "rescaled.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NOTE:** Since the the xarray object is in dask, the actual computation isn't performed until `.compute()`is called. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rescaled.values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Basic ML pipeline\n",
+    "More on this in the next user guide [Machine_Learning](./04_Machine_Learning.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask_ml.cluster import SpectralClustering\n",
+    "from dask.distributed import Client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client = Client(processes=False)\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X = client.persist(rescaled)"
```

Comment:
Maybe mention something about the shape of `X` for rescaled here and why it is now suitable for the ML pipeline.